### DISCUSSION TITLE
Arabic names inverted

### DISCUSSION THREAD

**Discussion by shajder** [pelicanmapping/osgearth#2231] [Support Q&A] [answered]
Hello

I am trying to display TFS feature point data (city names) with osgEarth. In my json file text looks correct:

![image](https://user-images.githubusercontent.com/11174520/222080697-69c121a8-9c10-4511-aa4d-9be1fb97ff30.png)

but it seems like one of osgEarth's 3rd party libraries tries to divert order of letters and as result I am getting this:

![image](https://user-images.githubusercontent.com/11174520/222081718-883cb488-6d52-4de5-b258-3d383a745629.png)

Feature style uses `text-encoding : utf-8` but I don't think that is the source of the problem. Where to start investigation ? Any clue greatly appreciated !!

Thanks
Marcin

**Comment by gwaldron** [comment, marked as answer]
Are you using an earth file? Try adding this to your style:
```json
text-layout: rtl;
```

**Reply by shajder** [reply]
Thanks! Indeed this helps with arabic names.
